### PR TITLE
vm: refactor Message and TxContext 

### PR DESCRIPTION
**Pull request by gabrocheleau**
Resolves https://github.com/ethereumjs/ethereumjs-monorepo/issues/528

### CODE REVIEW

**Comment by gabrocheleau** [comment]
An issue I ran into while attempting to improve the types is the following: https://github.com/ethereumjs/ethereumjs-monorepo/blob/vm/refactor-message-tx-context/packages/vm/src/evm/evm.ts#L481-L483 
```
    if (message.selfdestruct) {
      eei._result.selfdestruct = message.selfdestruct
    }
```
`message.selfdestruct ` used to be typed as `any`, but it only ever gets assigned with the following type: `selfdestruct?: { [k: string]: boolean }` (e.g. in `runCall`), but `eei._result.selfdestruct` has type `selfdestruct: { [k: string]: Buffer }`, where the Buffer is the Address where the selfdestructed address funds were sent. 

Should be assigning an Address to the `[k:string]` keys whose value is `true`?

**Comment by ryanio** [comment]
@gabrocheleau it looks like it sets it in eei _selfDestruct: https://github.com/ethereumjs/ethereumjs-monorepo/blob/019604cc95e377eaeb42c24710197088799d377b/packages/vm/src/evm/eei.ts#L415-L420

File: packages/vm/src/runCall.ts [hunk 1]
```diff
@@ -35,24 +35,24 @@ export interface RunCallOpts {
...
+  const txContext: TxContext = {
+    gasPrice: opts.gasPrice ?? BigInt(0),
+    origin: opts.origin ?? opts.caller ?? Address.zero(),
+  }
 
   const message = new Message({
     caller: opts.caller,
-    gasLimit: opts.gasLimit ?? 0xffffffn,
+    gasLimit: opts.gasLimit ?? BigInt(0xffffff),
     to: opts.to ?? undefined,
```
Comment (ryanio) [inline packages/vm/src/runCall.ts:46]: Don't need undefined

File: packages/vm/src/runCall.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export interface RunCallOpts {
   block?: Block
   gasPrice?: bigint
   origin?: Address
-  caller?: Address
```
Comment (ryanio) [inline packages/vm/src/runCall.ts:15]: This should still be optional, can just default to zero address in `new Message`

**Comment by holgerd77** [comment]
This will need some cherry-picked resubmission.

File: packages/vm/src/runCall.ts (hunk 1 above)
Comment (gabrocheleau) [reply packages/vm/src/runCall.ts:46]: Fixed!

File: packages/vm/src/runCall.ts (hunk 2 above)
Comment (gabrocheleau) [reply packages/vm/src/runCall.ts:15]: Fixed!

**Comment by gabrocheleau** [comment]
Closing in favor or this clean-up PR: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1819

**Pull request closed without merging**